### PR TITLE
Create interaction branch

### PR DESCRIPTION
**Pull request by sulaimonaa**
#Add clear all task features
#Add completed features
#Add on-click method for completed and clear features

### CODE REVIEW

File: modules/taskFunctions.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  completeTask(todoId) {
+    todoId -= 1;
+    this.toDoList = this.toDoList.map((task, index) => ({
+      ...task,
+      completedValue: index === todoId ? !task.completed : this.completed,
+    }));
+
+    this.displayTasks();
+  }
```
Comment (t-yanick) [inline modules/taskFunctions.js:48]: - [ ] Great job working on this milestone.
- When I click on the checkbox to make task completed, the status does not change in the localstorage to completed:True.
- Secondly, I cannot tick the checkbox for more that one task. When I click the second, the first deselects.
- Finally when I click on `Clear all completed` even the tasks where the checkbox is not ticked are deleted. It is not supposed function like this.
image

All these as per project requirements are supposed to work. Kindly look once one into these features

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @sulaimonaa , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  JavaScript file imported as module
:heavy_check_mark: Use of correct Github flow
:heavy_check_mark: Array method used

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @sulaimonaa 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - All changes requested have been implemented 🥇  


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kanyorok** [comment]
I have compared the code to best practices. You have a clean and quality flow of code 👍🏻. Keep up the good work. However, 😛 you could furnish the naming conventions so that they meet all the variable naming conventions of the code. Otherwise, good job 💯

**Pull request merged**